src/commands/remove.rs

```diff
@@ -0,0 +1,134 @@
+use aqora_runner::python::{PipOptions, PipPackage};
+use clap::Args;
+use indicatif::ProgressBar;
+use std::time::Duration;
+use tokio::fs;
+use toml_edit::DocumentMut;
+
+#[derive(Args, Debug)]
+#[command(author, version, about)]
+pub struct Remove {
+    pub deps: Vec<String>,
+}
+
+pub async fn remove(args: Remove, global: GlobalArgs) -> Result<()> {
+    let mut deps = Vec::new();
+    for dep in args.deps.iter() {
+        let req = Requirement::parse(dep, &global.project)
+            .map_err(|e| error::user(&format!("Invalid requirement '{dep}': {e}"), ""))?;
+        if !req.extras.is_empty() || req.version_or_url.is_some() || req.marker.is_some() {
+            return Err(error::user(
+                &format!("Invalid requirement '{dep}'"),
+                "Only the package name is allowed",
+            ));
+        }
+        deps.push(req);
+    }
+    let _ = read_pyproject(&global.project).await?;
+    let progress = ProgressBar::new_spinner();
+    progress.set_message("Initializing virtual environment");
+    progress.enable_steady_tick(Duration::from_millis(100));
+    let env = init_venv(&global.project, global.uv.as_ref(), &progress, global.color).await?;
+    let project_file = RevertFile::save(pyproject_path(&global.project))?;
+    let mut toml = fs::read_to_string(&project_file)
+        .await?
+        .parse::<DocumentMut>()?;
+    let project = if let Some(project) = toml.get_mut("project") {
+        project
+    } else {
+        progress.finish_with_message("No dependencies to remove");
+        return Ok(());
+    };
+    let dependencies = if let Some(dependencies) = project
+        .as_table_mut()
+        .ok_or_else(|| {
+            error::user(
+                "Invalid pyproject.toml",
+                "The 'project' section must be a table",
+            )
+        })?
+        .get_mut("dependencies")
+    {
+        dependencies
+    } else {
+        progress.finish_with_message("No dependencies to remove");
+        return Ok(());
+    };
+    let dependencies = dependencies.as_array_mut().ok_or_else(|| {
+        error::user(
+            "Invalid pyproject.toml",
+            "The 'dependencies' section must be an array",
+        )
+    })?;
+    let mut removed_deps = Vec::new();
+    for dep in deps.iter() {
+        let mut removed = false;
+        while let Some(index) = dependencies
+            .iter()
+            .map(|d| {
+                d.as_str()
+                    .ok_or_else(|| {
+                        error::user("Invalid pyproject.toml", "Dependencies must be strings")
+                    })
+                    .and_then(|s| {
+                        s.parse::<Requirement>().map_err(|err| {
+                            error::system(&format!("Could not parse dependencies: {err}"), "")
+                        })
+                    })
+            })
+            .collect::<Result<Vec<_>>>()?
+            .iter()
+            .position(|old_dep| old_dep.name == dep.name)
```

Comment:
I think there might be an extra `collect()` that could be optimized away, right?

Reply:
done